server/src/main/java/org/opensearch/search/aggregations/bucket/filterrewrite/RangeAggregatorBridge.java

```diff
@@ -51,8 +54,14 @@ protected boolean canOptimize(ValuesSourceConfig config, RangeAggregator.Range[]
         return false;
     }
 
-    protected void buildRanges(RangeAggregator.Range[] ranges) {
+    protected void buildRanges(RangeAggregator.Range[] ranges, SearchContext context) {
         assert fieldType instanceof NumericPointEncoder;
+        Query unwrap = Helper.unwrapIntoConcreteQuery(context.query());
+        if (!(unwrap instanceof MatchAllDocsQuery)) {
+            setRanges.accept(null);
+            return;
+        }
+
```

Comment:
This might be fine for the regression fix. Can you create an issue for including `PointRangeQuery`, `FieldExistsQuery` scenarios similar to `DateHistogram` aggregation?

```
        if (cq instanceof PointRangeQuery) {
            final PointRangeQuery prq = (PointRangeQuery) cq;
            final long[] indexBounds = getShardBounds(leaves, fieldName);
            if (indexBounds == null) return null;
            return getBoundsWithRangeQuery(prq, fieldName, indexBounds);
        } else if (cq instanceof MatchAllDocsQuery) {
            return getShardBounds(leaves, fieldName);
        } else if (cq instanceof FieldExistsQuery) {
            // when a range query covers all values of a shard, it will be rewrite field exists query
            if (((FieldExistsQuery) cq).getField().equals(fieldName)) {
                return getShardBounds(leaves, fieldName);
            }
        }
```

Reply:
Or actually, we can include test case for `PointRangeQuery`, `FieldExistsQuery` along with range aggregation on the same and different fields and validate that the optimization applies only for the same field case?

Reply:
Created. Will add this support in a follow up PR.
#15217